modules/tinymce/src/plugins/link/main/ts/core/Utils.ts

```diff
@@ -184,6 +185,27 @@ const unlink = (editor: Editor) => {
   });
 };
 
+const unwrapOptions = (data: LinkDialogOutput) => {
+  const { class: cls, href, rel, target, text, title } = data;
+
+  return Obj.filter({
+    class: cls.getOrNull(),
+    href,
+    rel: rel.getOrNull(),
+    target: target.getOrNull(),
+    text: text.getOrNull(),
+    title: title.getOrNull()
+  }, (v, k) => Type.isNull(v) === false);
+};
+
+const link = (editor: Editor, attachState: AttachState, data: LinkDialogOutput) => {
+  hasRtcPlugin(editor) ? editor.execCommand('createlink', false, unwrapOptions(data)) : linkDomMutation(editor, attachState, data);
```

Comment:
side note: if we trust `createlink` in the core, or moved the dom mutation into core, this wouldn't be necessary